rpc/database.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Athanor Labs (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package rpc
+
+import (
+	"math/big"
+	"net/http"
+
+	"github.com/athanorlabs/atomic-swap/common/types"
+	mcrypto "github.com/athanorlabs/atomic-swap/crypto/monero"
+	"github.com/athanorlabs/atomic-swap/db"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// RecoveryDB contains methods for retrieving swap recovery info from the database.
+type RecoveryDB interface {
+	GetContractSwapInfo(id types.Hash) (*db.EthereumSwapInfo, error)
+	GetSwapPrivateKey(id types.Hash) (*mcrypto.PrivateSpendKey, error)
+	GetCounterpartySwapPrivateKey(id types.Hash) (*mcrypto.PrivateSpendKey, error)
+}
+
+// DatabaseService ...
+type DatabaseService struct {
+	rdb RecoveryDB
+}
+
+// NewDatabaseService returns a new DatabaseService.
+func NewDatabaseService(rdb RecoveryDB) *DatabaseService {
+	return &DatabaseService{
+		rdb: rdb,
+	}
+}
+
+// GetContractSwapInfoRequest ...
+type GetContractSwapInfoRequest struct {
+	ID types.Hash `json:"id" validate:"required"`
+}
+
+// GetContractSwapInfoResponse ...
+type GetContractSwapInfoResponse struct {
+	StartNumber     *big.Int                   `json:"startNumber" validate:"required"`
```

Comment:
Can we document this? Is it the height to start scanning monero blocks at?

Reply:
it's documented here: https://github.com/AthanorLabs/atomic-swap/blob/master/db/types.go#L16 should i copy paste the comments?

Reply:
I forgot that we were storing that. `newSwapBlockNum` would be unambiguous, albeit 4 characters longer. How does the value get used? Would it make sense the store `newSwapTxID` instead?

Reply:
I don't really think the user needs this necessarily, but I'll change the name to `startBlockNumber`